### PR TITLE
Cards and bias files for ggH->tautau and VBF->tautau flat mass samples

### PR DESCRIPTION
**Pull request by danielwinterbottom**
- Adding cards and bias files needed to produce ggH->tautau and VBF->tautau MC samples with approximately flat mass distributions between 30-300 GeV
- these are needed to prodcue MC samples for a ditau mass regression
- Note I also had to change 1 line in the gridpack_generation.sh script to make sure the bias file was taken from the correct directory

### CODE REVIEW

File: bin/MadGraph5_aMCatNLO/gridpack_generation.sh [hunk 1]
```diff
@@ -206,7 +206,7 @@ make_gridpack () {
         echo "copying bias module folder. Current dir:"
         pwd
         ls -lrth
-        cp -r $CARDSDIR/BIAS/* $MGBASEDIRORIG/Template/LO/Source/BIAS
+        cp -r $CARDSDIR/BIAS/* Template/LO/Source/BIAS
```
Comment (bbilin) [inline bin/MadGraph5_aMCatNLO/gridpack_generation.sh:209]: Please do not change the general script. You can put a copy into your directory.

**Review by bbilin — COMMENTED** [review]
Can you please also provide some mass plots if you have any private validations?

File: bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/DiTauBiasedMass/ggToScalarToTauTau_BiasedMass_Mass30to300/ggToScalarToTauTau_BiasedMass_Mass30to300_run_card.dat [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+#
+#*********************************************************************
+# Store info for systematics studies                                 *
+# WARNING: Do not use for interference type of computation           *
+#*********************************************************************
+   True  = use_syst      ! Enable systematics studies
+#
+systematics = systematics_program ! none, systematics [python], SysCalc [depreceted, C++]
+['--mur=0.5,1,2', '--muf=0.5,1,2', '--pdf=errorset'] = systematics_arguments ! see: https://cp3.irmp.ucl.ac.be/projects/madgraph/wiki/Systematics#Systematicspythonmodule
+# Syscalc is deprecated but to see the associate options type'update syscalc'
```
Comment (bbilin) [inline bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/DiTauBiasedMass/ggToScalarToTauTau_BiasedMass_Mass30to300/ggToScalarToTauTau_BiasedMass_Mass30to300_run_card.dat:271]: These can be removed, we launch the syst variations elsewhere in the code.

Comment (bbilin) [reply bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/DiTauBiasedMass/ggToScalarToTauTau_BiasedMass_Mass30to300/ggToScalarToTauTau_BiasedMass_Mass30to300_run_card.dat:271]: Also for other two run card files.

Comment (danielwinterbottom) [reply bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/DiTauBiasedMass/ggToScalarToTauTau_BiasedMass_Mass30to300/ggToScalarToTauTau_BiasedMass_Mass30to300_run_card.dat]: OK the lines have been removed

**Comment by danielwinterbottom** [comment]
> Can you please also provide some mass plots if you have any private validations?
@bbilin sure here are some plots I produced for ggH and VBF
ggH_biased_mass
VBF_biased_mass

File: bin/MadGraph5_aMCatNLO/gridpack_generation.sh (hunk 1 above)
Comment (danielwinterbottom) [reply bin/MadGraph5_aMCatNLO/gridpack_generation.sh:209]: @bbilin Let me know how to proceed, whether to revert back this script or not

Comment (bbilin) [reply bin/MadGraph5_aMCatNLO/gridpack_generation.sh:209]: Thanks, but I do not understand why it fails with $MGBASEDIRORIG which is the directory  set as $(echo ${MG%$MG_EXT} | tr "." "_"). In line 190 of the current script you cd to this, and it is used everywhere. Is your Template directory under the MG folder? (I assume so)

 https://github.com/cms-sw/genproductions/blob/master/bin/MadGraph5_aMCatNLO/gridpack_generation.sh#L190

File: bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/DiTauBiasedMass/VBFToScalarToTauTau_BiasedMass_Mass30to300/VBFToScalarToTauTau_BiasedMass_Mass30to300_run_card.dat [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+ -1 = dynamical_scale_choice ! Choose one of the preselected dynamical choices
+ 1.0  = scalefact        ! scale factor for event-by-event scales
+#*********************************************************************
+# Type and output format
+#*********************************************************************
+  False     = gridpack  !True = setting up the grid pack
+  -1.0 = time_of_flight ! threshold (in mm) below which the invariant livetime is not written (-1 means not written)
+  3.0 = lhe_version       ! Change the way clustering information pass to shower.        
+  True = clusinfo         ! include clustering tag in output
+  average =  event_norm       ! average/sum. Normalization of the weight in the LHEF
```
[the remark this answers is not in the document]
Comment (bbilin) [reply bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/DiTauBiasedMass/VBFToScalarToTauTau_BiasedMass_Mass30to300/VBFToScalarToTauTau_BiasedMass_Mass30to300_run_card.dat:66]: Ok thanks for the explanation. Can you please comment next to the mass_bias.f the use case for these samples, for future reference?

[the remark this answers is not in the document]
Comment (danielwinterbottom) [reply bin/MadGraph5_aMCatNLO/cards/production/13p6TeV/DiTauBiasedMass/VBFToScalarToTauTau_BiasedMass_Mass30to300/VBFToScalarToTauTau_BiasedMass_Mass30to300_run_card.dat:66]: OK I have added some comments to the mass_bias.f files in this commit: https://github.com/cms-sw/genproductions/pull/3698/commits/2f7dad088a5864bb01e8caa64d53ad0d35e47413

File: bin/MadGraph5_aMCatNLO/gridpack_generation.sh (hunk 1 above)
Comment (bbilin) [reply bin/MadGraph5_aMCatNLO/gridpack_generation.sh:209]: OK I get it now. thanks.

**Pull request merged**